ui/src/ducks/app/salt.js

```diff
@@ -178,6 +181,13 @@ export function* refreshJobStatus(job) {
       }),
     );
   }
+
+  // Check if the job jid from localStorage is exist. If not, we should remove it from localStorage
+  const jobNotFound = result?.return?.find(
+    (job) => Object.values(job)[0]['Error'] === ERROR_NO_JOB_JID_FOUND,
+  );
+  yield call(removeJobFromLocalStorage, Object.keys(jobNotFound)[0]);
```

Comment:
Indeed, good catch! It will delete good jobs. 

I don't know how we can check the StartTime. Because let's say I executed Salt job from an instance, and switch to another instance. I will have a StartTime too. 

Maybe we should not remove any JID from localStorage?

Reply:
> Because let's say I executed Salt job from an instance, and switch to another instance. I will have a StartTime too.

Are you sure? If the job ID doesn't exist, then I don't see how Salt can give you a `StartTime`...

Reply:
@gdemonet That's weird that you get an error key with `Cannot contact returner or no job with this jid`, seems like something is wrong (the call failed or the ID really doesn't exists).
The presence of the `Error` key is also what we used in the operator:


```go
	// Check for "Error" field.
	if reason, found := info["Error"]; found {
		// Unknown Job ID: maybe the Salt server restarted or something like that.
		logger.Info("Salt job not found")
		return nil, fmt.Errorf("cannot get status for job %s: %v", jobID, reason)
	}
```
If your job is really in progress, then you'll get some info about it's status (and an key `Result` with an empty list since it's not finished yet):

```go
	// No result yet, the job is still running.
	if len(result) == 0 {
		logger.Info("Salt job is still running")
		return nil, nil
	}
```

At least for the states used by the storage-operator, I don't know if this holds for every Salt call.

Reply:
OK, indeed there is something wrong somewhere (my bet is on Salt).

Tested the following:

- for a runner calling our `deploy_node` orchestrate, as the UI does

```
# salt-run --async state.orch metalk8s.orchestrate.deploy_node saltenv=metalk8s-2.6.0-dev pillar='{orchestrate: {node_name: worker-1}}'
[WARNING ] Running in asynchronous mode. Results of this execution may be collected by attaching to the master event bus or by examing the master job cache, if configured. This execution is running under tag salt/run/20201201163643992851
# salt-run jobs.print_job 20201201163643992851
20201201163643992851:
    ----------
    Error:
        Cannot contact returner or no job with this jid
    Result:
        ----------
    StartTime:
        2020, Dec 01 16:36:43.992851
[INFO    ] Runner completed: 20201201163704623108
```

- for a simple `salt` command, as the storage-operator does:

```
# salt --async 'worker-1' state.sls metalk8s.volumes saltenv=metalk8s-2.6.0-dev
Executed command with job ID: 20201201164240351912
# salt-run jobs.print_job 20201201164240351912
20201201164240351912:
    ----------
    Arguments:
        - metalk8s.volumes
        |_
          ----------
          __kwarg__:
              True
          saltenv:
              metalk8s-2.6.0-dev
    Function:
        state.sls
    Minions:
        - worker-1
    Result:
        ----------
    StartTime:
        2020, Dec 01 16:42:40.351912
    Target:
        worker-1
    Target-type:
        glob
    User:
        root
[INFO    ] Runner completed: 20201201164247350096
```

Of course, if I wait long enough, I get a full `Result` for both jobs, meaning there is no actual error in the first case outside the missing reporter.

Reply:
> > Because let's say I executed Salt job from an instance, and switch to another instance. I will have a StartTime too.
> 
> Are you sure? If the job ID doesn't exist, then I don't see how Salt can give you a `StartTime`...

@gdemonet  I think if you try with the real job which generates the real timestamp (not a random number), you can get StartTime.

Reply:
OK. As discussed during the meeting, I will drop this commit. 

Since this issue will not happen in the production.